### PR TITLE
Feat: application_controller should be responsible for deleting application revision

### LINKED ISSUE #5710: [Feature] application_controller should be responsible for deleting application revision
**Is your feature request related to a problem? Please describe.**

When application is deleted, the application revision is deleted by Kubernetes following the OwnerReference mechanism. This could be slow when the system has high load. We should let application controller to handle it as well as a double check.

**Describe the solution you'd like**

Move pkg/controller/core.oam.dev/v1alpha2/application/apply.go:garbageCollection to pkg/resourcekeeper/gc.go:GarbageCollect and support cleaning up application revision when application is fully removed.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by dhanusaputra**
### Description of your changes

Fixes #5710 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by Somefive** [comment]
Is it still a draft PR or ready for review?

**Comment by dhanusaputra** [comment]
it's draft
@Somefive could you help quick look the PR? whether as expected or not, thanks

**Comment by Somefive** [comment]
@dhanusaputra Sorry for the late reply. Generally this is what we what, moving the revision cleanup function into GarbageCollect. There are several things you could look into
1. Try to keep `GarbageCollect(context.Context, ...GCOption) (bool, []v1beta1.ManagedResource, error)` unchanged. The reconcile client should be passed in through the resourceKeeper, not the function parameter.
2. DisableAllApplicationRevisionGCOption -> DisableApplicationRevisionGCOption
3. For DisableAllComponentRevisionGCOption, try to merge it to the existing component revision gc in resourceKeeper.

BTW, great job!

File: pkg/resourcekeeper/gc.go [hunk 1]
```diff
@@ -162,6 +172,13 @@ func (h *resourceKeeper) garbageCollect(ctx context.Context, cfg *gcConfig) (fin
 			return false, waiting, errors.Wrapf(err, "failed to garbage collect legacy resource trackers")
 		}
 	}
+
+	if !cfg.disableApplicationRevisionGC {
+		if err = gc.GarbageCollectApplicationRevisionResourceTrackers(ctx); err != nil {
```
Comment (Somefive) [inline pkg/resourcekeeper/gc.go:177]: GarbageCollectApplicationRevisionResourceTrackers -> GarbageCollectApplicationRevision?

File: pkg/resourcekeeper/gc.go [hunk 2]
```diff
@@ -162,6 +172,13 @@ func (h *resourceKeeper) garbageCollect(ctx context.Context, cfg *gcConfig) (fin
...
+			return false, waiting, errors.Wrapf(err, "failed to garbage collect application revision resource trackers")
```
Comment (Somefive) [inline pkg/resourcekeeper/gc.go:178]: failed to garbage collect application revision resource trackers -> failed to garbage collect application revision

**Review by Somefive — COMMENTED** [review]
Great job! Generally LGTM after nit. BTW, since the `gc.go` might be a bit large now, we can move the revision cleanup logic into `gc_rev.go`.

**Review by Somefive — APPROVED** [review]

**Pull request merged**